validation/hooks.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	tap "github.com/mndrix/tap-go"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validation/util"
+	uuid "github.com/satori/go.uuid"
+)
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+
+	var output string
+	config := util.LifecycleConfig{
+		Actions: util.LifecycleActionCreate | util.LifecycleActionStart | util.LifecycleActionDelete,
+		PreCreate: func(r *util.Runtime) error {
+			r.SetID(uuid.NewV4().String())
+			g := util.GetDefaultGenerator()
+			output = filepath.Join(r.BundleDir, g.Spec().Root.Path, "output")
+			prestart1 := rspec.Hook{
+				Path: filepath.Join(r.BundleDir, g.Spec().Root.Path, "/bin/sh"),
+				Args: []string{
+					"sh", "-c", fmt.Sprintf("echo 'pre-start1 called' >> %s", output),
+				},
+			}
+			g.AddPreStartHook(prestart1)
```

Comment:
No need for a `prestart1` temporary variable, you can just use:

```go
err := g.AddPreStartHook(rspec.Hook{
  …
})
if err != nil {
  return err
}
```

That also checks this call's returned error, which you're currently ignoring.